compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1005,10 +1004,14 @@ class LoweringConfigGenerator {
       SmallVector<int64_t> reductionTileSizes;
       splitParallelAndReductionTiles(rootOp, parallelTileSizes,
                                      reductionTileSizes);
-      appendLoweringConfigLevelAttr(items, TilingLevel::CacheParallelTiles,
-                                    parallelTileSizes);
-      appendLoweringConfigLevelAttr(items, TilingLevel::CacheReductionTiles,
-                                    reductionTileSizes);
+      if (llvm::any_of(parallelTileSizes, [](int64_t v) { return v != 0; })) {
+        appendLoweringConfigLevelAttr(items, TilingLevel::CacheParallelTiles,
+                                      parallelTileSizes);
+      }
```

Comment:
I'd be tempted to move 
```cpp
if (llvm::any_of(tileSizes, [](int64_t v) { return v != 0; })) {
```
inside `appendLoweringConfigLevelAttr`. Presumably, we will never require ``appendLoweringConfigLevelAttr` to add the attributes unconditionally?

Reply:
Yes, good catch. Only distribution can't be ignored, and I can add the logic.